automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala

```diff
@@ -338,6 +339,72 @@ class WorkspaceApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike
+
+            withMethod("writer-method", MethodData.SimpleMethod) { methodName =>
+              val method = MethodData.SimpleMethod.copy(methodName = methodName)
+
+              Rawls.methodConfigs.createMethodConfigInWorkspace(projectName, workspaceName,
+                method, method.methodNamespace, method.methodName, 1,
+                SimpleMethodConfig.inputs, SimpleMethodConfig.outputs, method.rootEntityType)(ownerAuthToken)
+
+              Orchestration.methods.setMethodPermissions(
+                method.methodNamespace,
+                method.methodName,
+                method.snapshotId,
+                studentA.email,
+                "OWNER"
+              )(ownerAuthToken)
+
+              val submissionId = Rawls.submissions.launchWorkflow(projectName, workspaceName, method.methodNamespace, method.methodName, method.rootEntityType, "participant1", "this", false)(studentAToken)
+              // make sure the submission has not errored out
+              eventually {
+                val submissionStatus = Rawls.submissions.getSubmissionStatus(projectName, workspaceName, submissionId)(studentAToken)._1
+                List("Accepted", "Evaluating", "Submitting", "Submitted") should contain (submissionStatus)
+              }
+            }(ownerAuthToken)
+          }(ownerAuthToken)
+        }
+      }
+    }
+
+    "should not allow writers" - {
+      "to launch workflows if they don't have can-compute permission" in {
+        withCleanBillingProject(owner) { projectName =>
+          withWorkspace(projectName, prependUUID("writer-cannot-launch-workflow"), aclEntries = List(AclEntry(studentA.email, WorkspaceAccessLevel.Writer, canCompute = Some(false)))) { workspaceName =>
+            Rawls.entities.importMetaData(projectName, workspaceName, entity)(ownerAuthToken)
+
+            withMethod("writer-method", MethodData.SimpleMethod) { methodName =>
+              val method = MethodData.SimpleMethod.copy(methodName = methodName)
+
+              Rawls.methodConfigs.createMethodConfigInWorkspace(projectName, workspaceName,
+                method, method.methodNamespace, method.methodName, 1,
+                SimpleMethodConfig.inputs, SimpleMethodConfig.outputs, method.rootEntityType)(ownerAuthToken)
+
+              Orchestration.methods.setMethodPermissions(
+                method.methodNamespace,
+                method.methodName,
+                method.snapshotId,
+                studentA.email,
+                "OWNER"
+              )(ownerAuthToken)
+
+              eventually {
+                val submissionException = intercept[RestException] {
+                  Rawls.submissions.launchWorkflow(projectName, workspaceName, method.methodNamespace, method.methodName, method.rootEntityType, "participant1", "this", false)(studentAToken)
+                }
+                assertExceptionStatusCode(submissionException, 403)
```

Comment:
I don't think this needs to be inside an eventually - we should be returned a 403 every time we try to launch this workflow. If not, the test should fail.